tree/ntuple/v7/src/RNTupleDescriptor.cxx

```diff
@@ -731,6 +743,52 @@ std::unique_ptr<ROOT::Experimental::RNTupleModel> ROOT::Experimental::RNTupleDes
 }
 
 
+////////////////////////////////////////////////////////////////////////////////
+
+ROOT::Experimental::RResult<void>
+ROOT::Experimental::RClusterDescriptorBuilder::AddPageRange(const RClusterDescriptor::RPageRange &pageRange)
+{
+   if (fCluster.fPageRanges.count(pageRange.fColumnId) > 0)
+      return R__FAIL("column ID conflict");
+   fCluster.fPageRanges[pageRange.fColumnId] = pageRange.Clone();
+   return RResult<void>::Success();
+}
+
+
+ROOT::Experimental::RResult<void>
+ROOT::Experimental::RClusterDescriptorBuilder::CommitColumnRange(
+   DescriptorId_t columnId, std::uint64_t firstElementIndex)
+{
+   RClusterDescriptor::RColumnRange columnRange{columnId, firstElementIndex, RClusterSize(0)};
+   auto iter = fCluster.fPageRanges.find(columnId);
+   if (iter != fCluster.fPageRanges.end()) {
+      for (const auto &pi : iter->second.fPageInfos) {
+         columnRange.fNElements += pi.fNElements;
+      }
+   }
+   fCluster.fColumnRanges[columnId] = columnRange;
+   return RResult<void>::Success();
+}
+
+
+ROOT::Experimental::RResult<ROOT::Experimental::RClusterDescriptor>
+ROOT::Experimental::RClusterDescriptorBuilder::MoveDescriptor()
+{
+   if (fCluster.fClusterId == kInvalidDescriptorId)
```

Comment:
On the surface those checks (in particular the loop) look "heavy" in a move function.   Shouldn't the caller verify (likely during the building up) that the builder is valid before calling this?  (i.e. reducing from "loop to add/build then loop again to check" to just "loop to check and add/build")

Reply:
Despite the name, this is really a "build" function, i.e. same as `RColumnDescriptorBuilder::MakeDescriptor()`.

I think that the intent was to make it clear that the internal `RClusterDescriptor` is not valid after calling this function.  In any case, we might rename it in a subsequent PR.